crates/script/src/dryrun.rs

```diff
@@ -0,0 +1,96 @@
+use alloy_consensus::Transaction;
+use alloy_primitives::{hex, utils::format_units};
+use forge_script_sequence::TransactionWithMetadata;
+use foundry_common::TransactionMaybeSigned;
+
+/// Format transaction details for display
+pub fn format_transaction_details(index: usize, tx: &TransactionWithMetadata) -> String {
+    let mut output = format!("\n### Transaction {index} ###\n\n");
+
+    // Contract info (to)
+    if let Some(addr) = tx.contract_address {
+        if let Some(name) = &tx.contract_name {
+            output.push_str(&format!("to: {name}({addr})\n"));
+        } else {
+            output.push_str(&format!("to: {addr}\n"));
+        }
+    } else {
+        output.push_str("to: <contract creation>\n");
+    }
+
+    // Transaction data
+    let (input, gas, gas_price, value, max_fee_per_gas, max_priority_fee_per_gas) = match tx.tx() {
```

Comment:
You should be able to use [UIFmt](https://github.com/foundry-rs/foundry/blob/aa0161e8c513b2a686439cc1b9a07e20a5701120/crates/common/fmt/src/ui.rs#L322) to format the transaction.